qemu/tests/virtio_console.py

```diff
@@ -117,6 +117,100 @@ def add_virtio_ports_to_vm(vm, params, serial_device):
         serial_id, serial_name, filename, backend))
 
 
+def _port_unplug(test, vm, ports, port_idx, intr_time=0):
```

Comment:
I won't recommend to reuse these functions, it is unnecessary to record those properties between unplug and plug, the device object is not deleted, it is only removed from vm.devices, we might hotplug the device(the QDevice object) directly.